### PR TITLE
Improve event filters

### PR DESCRIPTION
**Pull request by michael-e**
This PR improves event filters as described in #303.

Additionally, the last commit (02aac90) adds additional constants to the `MemberEventMessages` class and implements error codes for all Event Filters. Please note that adding these error codes to Filters (as opposed to Events) might be controversial, so I am open to feedback!

### CODE REVIEW

File: extension.driver.php [hunk 1]
```diff
@@ -703,7 +703,7 @@ public function appendFilter($context) {
 				);
 			}
 
-			if(!FieldManager::isFieldUsed(self::getFieldType('authentication'))) {
+			if(FieldManager::isFieldUsed(self::getFieldType('authentication'))) {
```
Comment (nitriques) [inline extension.driver.php]: This does not make it show "always", but only if a authentication field is used. It was a bug ?

File: lib/member.symphony.php [hunk 2]
```diff
@@ -372,6 +372,80 @@ public function filter_LockActivation(array &$context) {
...
+					__('Member not found.'),
+					array(
+						'message-id' => MemberEventMessages::MEMBER_INVALID
+					)
+				);
+
+				return;
+			}
+
+			if (is_null($this->section->getFieldHandle('authentication'))) {
```
Comment (nitriques) [inline lib/member.symphony.php]: We could do ! instead of is_null ?

File: lib/member.symphony.php [hunk 3]
```diff
@@ -372,6 +372,80 @@ public function filter_LockActivation(array &$context) {
...
+						'message-id' => MemberEventMessages::MEMBER_ERRORS
+					)
+				);
+
+				return;
+			}
+
+			$member_id = null;
+			if (isset($_POST['fields'][$this->section->getFieldHandle('authentication')]['validate'])) {
+				$password = Symphony::Database()->cleanValue(
```
Comment (nitriques) [inline lib/member.symphony.php]: We do not need to clean the password, as it will be hashed anyways.

File: extension.driver.php (hunk 1 above)
Comment (michael-e) [reply extension.driver.php]: Good catch! Yes, I can only suspect that this was a bug, introduced by accident. The filter will call the `filter_UpdatePasswordLogin` function, which is completely useless without an authentication field. So the filter itself has been useless all the time, and I don't consider this a breaking change but a bugfix. (Please, confirm! I don't want to break stuff, of course.)

For the same reason showing the filter is only useful if an authentication field exists, so IMHO "always if the field exists" is enough.

File: lib/member.symphony.php (hunk 2 above)
Comment (michael-e) [reply lib/member.symphony.php]: By design `getFieldHandle` will return `null` or a string, so why do you prefer `!`?

I'd prefer to be consistent with the rest of the code. If you search the extension for `$this->section->getFieldHandle`, you will see that `is_null` is always used to test the return value.

Comment (michael-e) [reply lib/member.symphony.php]: So if we need to change how these return values are tested, I suggest to open another issue.

File: lib/member.symphony.php (hunk 3 above)
Comment (michael-e) [reply lib/member.symphony.php]: Yes, you are right.

Comment (michael-e) [reply lib/member.symphony.php]: See 3216572

File: extension.driver.php (hunk 1 above)
Comment (nitriques) [reply extension.driver.php]: LGTM then!

File: lib/member.symphony.php (hunk 2 above)
Comment (nitriques) [reply lib/member.symphony.php]: Na it's ok.

File: extension.driver.php (hunk 1 above)
Comment (michael-e) [reply extension.driver.php]: BTW, I found the misunderstanding. In https://github.com/symphonycms/members/issues/44#issuecomment-15017336 @jensscherbl said:

> Added this as a filter (Members: Login), which is only available if no activation is needed. Just like @brendo suggested.

So the idea was to only show the filter if no activation field exists. But actually he tested for the authentication field. I don't know why nobody ever noticed this.

(I don't think that we need to test for the activation field. It's up to the developer to use the the filter (and see if it works). The filter just attempts what the name says: Log the Member in.)

**Comment by michael-e** [comment]
I will wait a bit with merging, because:

1. Maybe @brendo joins the party.
2. An [interesting aspect](https://github.com/symphonycms/members/issues/303#issuecomment-342992010) has popped up.

File: extension.driver.php (hunk 1 above)
Comment (nitriques) [reply extension.driver.php]: Good.

**Comment by michael-e** [comment]
IMHO this can be pulled now. #303 can be closed then. I suggest to release 1.10.0, because we have new features.

**Pull request merged**